File: examples/java/src/main/java/TokenizerExample.java
import com.tokenizers.Tokenizer;
import com.tokenizers.Encoding;

/**
 * Example demonstrating how to use the tokenizers-sys Java bindings
 */
public class TokenizerExample {
    
    public static void main(String[] args) {
        String testText = "The quick brown fox jumps over the lazy dog.";
        
        System.out.println("=== Tokenizers-sys Java Example ===");
        System.out.println("Text to tokenize: " + testText);
        
        // Test BERT tokenizer
        testTokenizer("bert-base-cased", testText);
        
        // Test GPT-2 tokenizer
        testTokenizer("gpt2", testText);
    }
    
    private static void testTokenizer(String modelName, String text) {
        System.out.println("\n=== Testing " + modelName + " ===");
        
        try (Tokenizer tokenizer = Tokenizer.fromPretrained(modelName)) {
            // Encode the text
            try (Encoding encoding = tokenizer.encode(text, true)) {
                // Get encoding information
                int[] ids = encoding.getIds();
                String[] tokens = encoding.getTokens();
                int[] typeIds = encoding.getTypeIds();
                int[] specialTokensMask = encoding.getSpecialTokensMask();
                int[] attentionMask = encoding.getAttentionMask();
                
                System.out.println("Number of tokens: " + encoding.getLength());
                System.out.println("\nTokens:");
                for (int i = 0; i < tokens.length; i++) {
                    System.out.printf("  %s (ID: %d, Type: %d, Special: %d, Attention: %d)%n",
                        tokens[i], ids[i], typeIds[i], specialTokensMask[i], attentionMask[i]);
                }
                
                // Decode back to text
                String decoded = tokenizer.decode(ids, true);
                System.out.println("\nDecoded text: " + decoded);
                
                // Verify roundtrip
                if (text.trim().equals(decoded.trim())) {
                    System.out.println("✓ Roundtrip successful!");
                } else {
                    System.out.println("✗ Roundtrip failed!");
                    System.out.println("  Original: '" + text + "'");
                    System.out.println("  Decoded:  '" + decoded + "'");
                }
            }
        } catch (Exception e) {
            System.err.println("Error testing " + modelName + ": " + e.getMessage());
            e.printStackTrace();
        }
    }
}

File: examples/kotlin/src/main/kotlin/TokenizerExample.kt
import com.tokenizers.Tokenizer
import com.tokenizers.Encoding

fun main() {
    val testText = "The quick brown fox jumps over the lazy dog."
    
    println("=== Tokenizers-sys Kotlin Example ===")
    println("Text to tokenize: $testText")
    
    // Test BERT tokenizer
    testTokenizer("bert-base-cased", testText)
    
    // Test GPT-2 tokenizer
    testTokenizer("gpt2", testText)
}

private fun testTokenizer(modelName: String, text: String) {
    println("\n=== Testing $modelName ===")
    
    try {
        Tokenizer.fromPretrained(modelName).use { tokenizer ->
            // Encode the text
            tokenizer.encode(text, true).use { encoding ->
                // Get encoding information
                val ids = encoding.getIds()
                val tokens = encoding.getTokens()
                val typeIds = encoding.getTypeIds()
                val specialTokensMask = encoding.getSpecialTokensMask()
                val attentionMask = encoding.getAttentionMask()
                
                println("Number of tokens: ${encoding.length}")
                println("\nTokens:")
                for (i in tokens.indices) {
                    println("  ${tokens[i]} (ID: ${ids[i]}, Type: ${typeIds[i]}, Special: ${specialTokensMask[i]}, Attention: ${attentionMask[i]})")
                }
                
                // Decode back to text
                val decoded = tokenizer.decode(ids, true)
                println("\nDecoded text: $decoded")
                
                // Verify roundtrip
                if (text.trim() == decoded.trim()) {
                    println("✓ Roundtrip successful!")
                } else {
                    println("✗ Roundtrip failed!")
                    println("  Original: '$text'")
                    println("  Decoded:  '$decoded'")
                }
            }
        }
    } catch (e: Exception) {
        System.err.println("Error testing $modelName: ${e.message}")
        e.printStackTrace()
    }
}
